js/state.js

```diff
@@ -0,0 +1,21 @@
+import { getToday } from "./dateManipulation.js"
+
+const State = function( data ){
+    this.value = data
+    this.prev = null
+    this.listeners = []
+}
+
+State.prototype.addListener = function( cb ){
+    this.listeners.push( cb )
+}
+
+State.prototype.setState = function(value){
+    this.prev = this.value
+    this.value = value
+    this.listeners.forEach( callback => callback( this.value, this.prev ))
+}
+
+export const selectedDate = new State( getToday())
+export const selectedMonth = new State(getToday())
```

Comment:
Suggest using some automatic code formatter. Notice you use spaces inside brackets in one place while you do not in another place. We normally solve such things with automatic formatters such as Prettier in our IDEs.